src/main/java/io/wcm/caconfig/extensions/persistence/impl/PersistenceUtils.java

```diff
@@ -281,6 +282,34 @@ public static void commit(ResourceResolver resourceResolver, String relatedResou
     }
   }
 
+  /**
+   * Checks if the given item is modified or newly added by comparing its properties with the current state of the resource.
+   *
+   * @param resolver The ResourceResolver to access the resource.
+   * @param resourcePath The path of the resource to compare against.
+   * @param item The ConfigurationPersistData item containing the properties to compare.
+   * @return true if the resource does not exist or if any property value differs, false otherwise.
+   */
+  public static boolean isItemModifiedOrNewlyAdded(ResourceResolver resolver, String resourcePath, ConfigurationPersistData item) {
+    Resource resource = resolver.getResource(resourcePath);
+    if (resource == null) {
+      return true; // Resource does not exist, so it is considered modified
+    }
+
+    ValueMap currentProperties = resource.getValueMap();
+    Map<String, Object> itemProperties = item.getProperties();
+
+    for (Map.Entry<String, Object> entry : itemProperties.entrySet()) {
```

Comment:
this check ignores one edge case: assume a property was removed from the new properties set - in that case the property has to be removed from the existing resource as well, so this changes goes unnoticed by this method. so you need to check also the set of property names of the existing resource and the new properties list, filtering out the ignored properties as it is done in the `replaceProperties` method.

this edge case should also be covered in the unit test.

Reply:
alternatively, i think you can compare two maps directly using their equals method - but make sure to use filtered maps removing the ignored properties. still, the edge cases should be covered by unit test.

Reply:
Updated the code with the edge case covered now Thanks